src/Agent.Sdk/Knob.cs

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Reflection;
+using Microsoft.VisualStudio.Services.Agent.Util;
+
+namespace Agent.Sdk
+{
+    public enum KnobSource
+    {
+        BuiltinDefault,
+        EnvironmentVariable,
+        RuntimeVariable,
+    };
+
+    public class KnobValue
+    {
+        public KnobSource Source { get;  private set;}
+        public string Which { get; private set; }
+        private string _value;
+        public KnobValue(string value, KnobSource source, string which=null)
+        {
+            _value = value;
+            Source = source;
+            Which = which;
+        }
+
+        public string AsString()
+        {
+            return _value;
+        }
+
+        public bool AsBoolean()
+        {
+            return StringUtil.ConvertToBoolean(_value);
+        }
+    }
+
+    public class AgentKnobs : KnobPanel
+    {
+        public static readonly IKnob UseNode10 = new Knob()
+        {
+            Name="UseNode10",
+            EnvironmentVariableNames=new List<string>{ "AGENT_USE_NODE10" },
+            RuntimeVariableNames=new List<string>{ "AGENT_USE_NODE10" },
+            Description="Forces the agent to use Node 10 handler for all Node-based tasks",
+            DefaultValue="false"
+        };
+    }
+
+    public class KnobPanel
+    {
+        public class Knob : IKnob
+        {
+            public string Name { get; set; }
+            public List<string> EnvironmentVariableNames { get; set; }
+            public List<string> RuntimeVariableNames { get; set; }
+            // public TBDType CommandLineOption {get; private set;}
+            public string Description { get; set; }
+            public bool IsDeprecated {get; set; } = false;  // is going away at a future date
+            public bool IsExperimental {get; set; } = false; // may go away at a future date
+            public string DefaultValue {get; set; } = "";
+
+            public KnobValue GetValue(IKnobValueContext context)
+            {
+                //TODO: First thing we need to do is check if deprecated or experimental and throw an event that
+                //      can be logged
+
+                // proposed order of operations is
+                // 1. Run time variables
+                // 2. Command Line Options (Not Yet Implemented)
+                // 3. Environment Variables
+                // 4. Default
+                if (context != null)
+                {
+                    if (RuntimeVariableNames != null)
```

Comment:
`if (RuntimeVariableNames is object)` or `if (!(RuntimeVariableNames is null))`. (`!=` can be overridden to do bad things; the `is` check always does the right thing.)